src/yarn/gather-versions.task.ts

```diff
@@ -3,23 +3,17 @@ import { join } from 'node:path';
 import { TaskOptions, TaskStep } from 'projen';
 import { TypeScriptWorkspace } from './typescript-workspace';
 
-export enum VersionMatch {
-  MAJOR = 'MAJOR',
-  MINOR = 'MINOR',
-  EXACT = 'EXACT',
-}
-
 export class GatherVersions implements TaskOptions, TaskStep {
   public get exec(): string {
-    const main = `require(path.join(path.dirname(require.resolve('${currentPackageName()}')), 'yarn', 'gather-versions.exec.js'))`;
-    return `node -e "${main}" ${this.project.name} ${this.versionMatch} --deps ${this.project
+    const main = `require(path.join(path.dirname(require.resolve('${currentPackageName()}')), 'yarn', 'gather-versions.exec.js')).cliMain()`;
```

Comment:
not strongly against this, but the general convention in projen has been that scripts are self-executing.

Reply:
I just thought this'd make it easy to test.